cmd/promtool/backfill_test.go

```diff
@@ -58,12 +59,13 @@ func queryAllSeries(t testing.TB, q storage.Querier, expectedMinTime, expectedMa
 	return samples
 }
 
-func testBlocks(t *testing.T, db *tsdb.DB, expectedMinTime, expectedMaxTime int64, expectedSamples []backfillSample, expectedNumBlocks int) {
+func testBlocks(t *testing.T, db *tsdb.DB, expectedMinTime, expectedMaxTime, expectedBlockDuration int64, expectedSamples []backfillSample, expectedNumBlocks int) {
 	blocks := db.Blocks()
-	require.Equal(t, expectedNumBlocks, len(blocks))
+	require.Equal(t, expectedNumBlocks, len(blocks), "did create correct number of blocks")
 
-	for _, block := range blocks {
-		require.Equal(t, true, block.MinTime()/tsdb.DefaultBlockDuration == (block.MaxTime()-1)/tsdb.DefaultBlockDuration)
+	for i, block := range blocks {
+		span := block.MaxTime() - block.MinTime()
+		require.Less(t, span, expectedBlockDuration, "block %d spans more time than the expected block duration", i)
```

Comment:
It should be like before
```go
require.Equal(t, true, block.MinTime()/expectedBlockDuration == (block.MaxTime()-1)/expectedBlockDuration)
```

Reason: the block min and max times are aligned to the block duration, i.e. for a block duration 2h, a block cannot span a 2h boundary, for example `mint = 1h30m, maxt = 2h30m` is not allowed in TSDB (technically allowed, but causes issues).

Comparing after division makes sure both min and max times are in the same block bucket.

Reply:
(indirectly it also makes sure it does not span beyond the block duration)

Reply:
OK. I really, really do not want to expect a boolean `true` - it's impossible to read and understand when you break it. That makes sense about the alignment though.